cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -651,16 +651,11 @@ void reader::impl::allocate_nesting_info()
   page_nesting_decode_info.host_to_device(_stream);
 }
 
-void reader::impl::load_and_decompress_data(std::vector<row_group_info> const& row_groups_info,
-                                            size_type num_rows)
+std::pair<bool, std::vector<std::future<void>>> reader::impl::create_and_read_column_chunks(
+  std::vector<row_group_info> const& row_groups_info, size_type num_rows)
 {
-  // This function should never be called if `num_rows == 0`.
-  CUDF_EXPECTS(num_rows > 0, "Number of reading rows must not be zero.");
```

Comment:
So no more checking for `num_rows`? If `num_rows == 0` is acceptable now, should we short circuit this?

Reply:
Nevermind, I just see this check below.